thorlcr/thorutil/thmem.cpp

```diff
@@ -1363,6 +1368,7 @@ void CThorSpillableRowArray::kill()
 {
     clearRows();
     CThorExpandingRowArray::kill();
+    stats.reset();
```

Comment:
I don't think this is wrong, but it does mean the semantics have change for all CThorSpillableRowArray contexts.

Previously CThorSpillableRowArray used to build up stat values over multiple runs (e.g. if reused because in a child query).
Now because it is reset here, and will rely on the context to ensure the stats from it are harvested before it is reset to avoid them being lost.
That now happens in e.g. lookupjoin's rhsRows case now - but are other use cases now incorrect (lose stats)?
e.g. CSpillableStream::nextRow calls rows.kill() - will it lose stats?

Reply:
I'll address this separately in: https://hpccsystems.atlassian.net/browse/HPCC-32568